src/oob/exchange.rs

```diff
@@ -136,6 +128,146 @@ impl From<x509::ValidationError> for PublisherRequestError {
     }
 }
 
+//------------ RepositoryResponse --------------------------------------------
+
+/// Type representing a <repository_response/>
+///
+/// This is the response sent to a CA by the publication server. It contains
+/// the details needed by the CA to send publication messages to the server.
+///
+/// See https://tools.ietf.org/html/rfc8183#section-5.2.4
+#[derive(Debug)]
+pub struct RepositoryResponse {
+    /// The optional 'tag' identifier used like a session identifier
+    tag: Option<String>,
+
+    /// The name the publication server decided to call the CA by.
+    /// Note that this may not be the same as the handle the CA asked for.
+    publisher_handle: String,
+
+    /// The Publication Server Identity Certificate
+    id_cert: IdCert,
+
+    /// The URI where the CA needs to send its publication messages
+    service_uri: uri::Http,
+
+    /// The Rsync base directory for objects published by the CA
+    sia_base: uri::Rsync,
+
+    /// The HTTPS notification URI that the CA can use
+    rrdp_notification_uri: uri::Http
+}
+
+impl RepositoryResponse {
+
+    /// Parses a <repository_response /> message.
+    pub fn decode<R>(reader: R) -> Result<Self, RepositoryResponseError>
+        where R: io::Read {
+
+        XmlReader::decode(reader, |r| {
+            r.take_named_element("repository_response", |a, r| {
+                if let Ok("1") = a.get_req("version") {
+                    // Ok, proceed
+                } else {
+                    return Err(RepositoryResponseError::InvalidVersion)
+                }
+
+                let tag = a.get_opt("tag");
+                let publisher_handle = a.get_req("publisher_handle")?;
+                let service_uri = uri::Http::from_bytes(
```

Comment:
You should probably use `HttpUri::from_slice` here.

Reply:
I left it like it is because `HttpUri::from_slice` just delegates to `HttpUri::from_bytes`. So, calling from `HttpUri::from_slice` would require an extra conversion (.as_ref() on the &str).